src/main/java/io/github/cslysy/interview/refactor/Refactor02.java

```diff
@@ -1,22 +1,34 @@
 package io.github.cslysy.interview.refactor;
 
+import org.assertj.core.api.Assertions;
 import org.junit.Test;
 
 import java.util.ArrayList;
 import java.util.Arrays;
+import java.util.Collection;
 import java.util.List;
 
+import static org.assertj.core.api.Assertions.assertThat;
+
 public class Refactor02 {
 
     private static List<String> namesToUpperCase(final List<List<String>> families) {
         final List<String> result = new ArrayList<>();
-        for (List<String> family : families) {
-            for (String names : family) {
-                if (names != null && !names.isEmpty()) {
-                    result.add(names.toUpperCase());
-                }
-            }
-        }
+
+        families.stream()
+                .flatMap(Collection::stream)
+                .filter(x -> x != null)
+                .filter(x -> x != "")
+                .forEach(x -> result.add(x.toUpperCase()));
```

Comment:
Generalnie o to chodziło, rozwiązanie jest poprawne z wyjątkiem forEach'a. Użyj tam proszę metody collect i nie przypisuj wyniku do zmiennej `result` tylko zwróć dane bez tworzenia niepotrzebnej zmiennej.